File: bddtests/src/main/kotlin/mocking/vision/models/appointments/Location.kt
package mocking.vision.models.appointments

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlElement

@XmlAccessorType(XmlAccessType.FIELD)
data class Location(
        @XmlAttribute(name = "id") var id: Int?= null,
        @XmlElement(namespace = "urn:vision")
        var name: String
)

File: backendworker/NHSOnline.Backend.GpSystems/Suppliers/Vision/Models/Location.cs
using System.Xml.Serialization;

namespace NHSOnline.Backend.GpSystems.Suppliers.Vision.Models
{
    public class Location
    {
        [XmlAttribute(AttributeName = "id")]
        public string Id { get; set; }

        [XmlElement(ElementName = "name")]
        public string Name { get; set; }
    }
}